### PR TITLE
Ben/factor out hparams

### PR DESCRIPTION
**Pull request by benleetownsend**

### CODE REVIEW

File: finetune/lm_base.py [hunk 1]
```diff
@@ -45,11 +42,11 @@ class LanguageModelBase(object, metaclass=ABCMeta):
...
     """
 
-    def __init__(self, autosave_path=None, max_length=MAX_LENGTH, verbose=True):
+    def __init__(self, hparams=None, autosave_path=None, max_length=None, verbose=True):
+        self.hparams = hparams or get_default_hparms()
         self.autosave_path = autosave_path or tempfile.mkdtemp()
         _LOGGER.info("Writing intermediate checkpoints to {}".format(self.autosave_path))
-        self.max_length = max_length
-        self.autosave_path = autosave_path
+        self.max_length = max_length or self.hparams.MAX_LENGTH
```
Comment (madisonmay) [inline finetune/lm_base.py]: Feels redundant, there shouldn't be two correct ways to pass the same value.  I would just remove the max_length argument from top level and do self.max_length = self.hparams.MAX_LENGTH if you want to alias things

Comment (madisonmay) [reply finetune/lm_base.py]: For other endpoints it makes sense to have an override.  But for the `__init__` method I don't think it makes sense.

File: finetune/lm_base.py [hunk 2]
```diff
@@ -100,17 +97,18 @@ def get_target_encoder(self):
         else:
             return RegressionEncoder()
 
-    def _finetune(self, *Xs, Y, batch_size=BATCH_SIZE, val_size=0.05, val_interval=150, val_window_size=5):
+    def _finetune(self, *Xs, Y, batch_size=None, val_size=0.05, val_interval=150, val_window_size=5):
```
Comment (madisonmay) [inline finetune/lm_base.py]: val_size and val_interval and val_window_size should have corresponding hparam options

**Comment by madisonmay** [comment]
The one other stylistic one is one that's no fun to change but one I'll bring up anyhow.  I don't think it makes sense for config options to be all caps any more now that they're arguments passed to a fn and not constants defined in a file.  Thoughts?  Worth changing or fuck it?

**Comment by madisonmay** [comment]
Oh god I feel so bad but I'm going to be that guy.  snake_case is the go to rather than camelCase for parameters.

**Comment by benleetownsend** [comment]
@madisonmay Check this out then ill merge.

**Pull request merged**